api/src/main/java/com/oviva/spicegen/api/CheckPermissions.java

```diff
@@ -0,0 +1,21 @@
+package com.oviva.spicegen.api;
+
+import com.oviva.spicegen.api.internal.CheckPermissionsImpl;
+import java.util.List;
+
+public interface CheckPermissions {
```

Comment:
```suggestion
public interface CheckBulkPermissions {
```
I'd keep the naming a bit more distinct, easier to read :)
Also better aligned with the underlying spicedb API.